### ISSUE TITLE
Test fails on macos

### ISSUE THREAD

**Issue by t20100** [silx-kit/silx#3317]
Some tests fails on macos:
```
======================================================================
FAIL: testArcRoiSwitchMode (silx.gui.plot.tools.test.testROI.TestRegionOfInterestManager)
Make sure we can switch mode by clicking on the ROI
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/tvincent/src/silx/build/lib.macosx-10.9-x86_64-3.7/silx/gui/plot/tools/test/testROI.py", line 653, in testArcRoiSwitchMode
    self.assertIs(item.getInteractionMode(), roi_items.ArcROI.PolarMode)
AssertionError: <silx.gui.plot.items._roi_base.RoiInteractionMode object at 0x11497e748> is not <silx.gui.plot.items._roi_base.RoiInteractionMode object at 0x11497e780>

======================================================================
FAIL: testLineInteraction (silx.gui.plot.tools.test.testROI.TestRegionOfInterestManager)
This test make sure that a ROI based on handles can be edited with
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/tvincent/src/silx/build/lib.macosx-10.9-x86_64-3.7/silx/gui/plot/tools/test/testROI.py", line 570, in testLineInteraction
    self.assertNotEqual(points[0, 1], result[0, 1])
AssertionError: 0.0 == 0.0
```

`testArcRoiSwitchMode` looks to be due to high-DPI screen and widget coordinates, still to see if is the test or something that needs a fix.

**Comment by t20100** [comment]
`testArcRoiSwitchMode` issue due to issue in high-dpi support in `PlotWidget` matplotlib backend (fine with the OpenGL one). This needs to be fixed.

**Comment by t20100** [comment]
Related to #2914

**Comment by t20100** [comment]
closed by #3318
